Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -861,6 +863,11 @@ private bool ShouldSandboxedProcessExecuteExternal
 
             info.RedirectedTempFolders = m_tempFolderRedirectionForVm.Select(kvp => (kvp.Key.ToString(m_pathTable), kvp.Value.ToString(m_pathTable))).ToArray();
 
+            if (m_sandboxConfig.AdminRequiredProcessExecutionMode.ExecuteExternalVm())

```

Comment:
Why don't use ShouldSandboxedProcessExecuteInVm here? Does "ShouldSandboxedProcessExecuteExternal" matter when translate Unc drive?

Reply:
This method is called when `ShouldSandboxedProcessExecuteExternal` is true. Then, we only need to translate host UNC drive if in fact we're about to run the pip in VM. If not, then the pip will only run on the host but not as a child process of BuildXL. Because the pip still runs on the host, there's no need to translate host UNC drive.

---
In reply to: [292239677](https://github.com/microsoft/BuildXL/pull/439#discussion_r292239677) [](ancestors = 292239677)